kloppy/infra/serializers/event/statsbomb/deserializer.py

```diff
@@ -216,17 +216,39 @@ def create_team(lineup, ground_type):
         return [home_team, away_team]
 
     def create_periods(self, raw_events):
```

Comment:
I think the implementation of this method could be simplified.

**Example**
```python
    def create_periods(self, raw_events):
        half_start_events = {}
        half_end_events = {}
        for event in raw_events.values():
            event_type = SB.EVENT_TYPE(event.raw_event["type"])
            period = event.raw_event["period"]

            if event_type == SB.EVENT_TYPE.HALF_START:
                half_start_events[period] = event.raw_event
            elif event_type == SB.EVENT_TYPE.HALF_END:
                half_end_events[period] = event.raw_event

        periods = []
        for start_event, end_event in zip_longest(
            half_start_events.values(), half_end_events.values()
        ):
            if start_event is None or end_event is None:
                raise DeserializationError(
                    "Failed to determine start and end time of periods."
                )
            start_timestamp = (
                periods[-1].end_timestamp
                + parse_str_ts(start_event["timestamp"])
                if len(periods) > 0
                else parse_str_ts(start_event["timestamp"])
            )
            end_timestamp = (
                start_timestamp + parse_str_ts(end_event["timestamp"])
                if end_event
                else None
            )
            period = Period(
                id=int(start_event["period"]),
                start_timestamp=start_timestamp,
                end_timestamp=end_timestamp,
            )
            periods.append(period)
        return periods
```

Reply:
Thanks, looks indeed more elegant. Made the requested changes.